src/dbnode/storage/series/lookup/entry.go

```diff
@@ -161,6 +202,61 @@ func (entry *Entry) OnIndexFinalize(blockStartNanos xtime.UnixNano) {
 	entry.DecrementReaderWriterCount()
 }
 
+// Write writes a new value.
+func (entry *Entry) Write(
+	ctx context.Context,
+	timestamp time.Time,
+	value float64,
+	unit xtime.Unit,
+	annotation []byte,
+	wOpts series.WriteOptions,
+) (bool, series.WriteType, error) {
+	if err := entry.maybeIndex(timestamp); err != nil {
+		return false, 0, err
+	}
+	return entry.Series.Write(
+		ctx,
+		timestamp,
+		value,
+		unit,
+		annotation,
+		wOpts,
+	)
+}
+
+// LoadBlock loads a single block into the series.
+func (entry *Entry) LoadBlock(
+	block block.DatabaseBlock,
+	writeType series.WriteType,
+) error {
+	// TODO(bodu): We can remove this once we have index snapshotting as index snapshots will
```

Comment:
this is preserving the existing behavior of indexing when loading while we roll out the boostrapping changes?

Reply:
Yes, commit logs are dropped when snapshots complete so w/o indexing when loading data blocks we will be missing series.